### PR TITLE
zlib-ng: provide better compatibility with zlib

### PR DESCRIPTION
**Pull request by Nekto89**
Specify library name and version:  **zlib-ng/all**

This change will allow zlib-ng to have same file/target names as zlib.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zlib-ng/all/conanfile.py [hunk 1]
```diff
@@ -106,6 +108,12 @@ def package_info(self):
             self.cpp_info.libs = [f"z{suffix}"]
         if self.options.zlib_compat:
             self.cpp_info.defines.append("ZLIB_COMPAT")
+            #copied from zlib
```
Comment (uilianries) [inline recipes/zlib-ng/all/conanfile.py:111]: I liked it, but it's not supported by the upstream.

I opened a PR to the upstream asking about it: https://github.com/zlib-ng/zlib-ng/pull/1601

Let's see what is the answer.

Comment (Nekto89) [reply recipes/zlib-ng/all/conanfile.py:111]: change is available in latest 2.1.6. Should I add this version in this PR?

Comment (uilianries) [reply recipes/zlib-ng/all/conanfile.py:111]: Yes, please.

Comment (Nekto89) [reply recipes/zlib-ng/all/conanfile.py:111]: 2.1.6 was added in other PR. I've rebased my branch on latest master.

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**